src/main/java/org/dependencytrack/persistence/ProjectQueryManager.java

```diff
@@ -538,6 +547,7 @@ public Project updateProject(Project transientProject, boolean commitIndex) {
         project.setDescription(transientProject.getDescription());
         project.setVersion(transientProject.getVersion());
         project.setClassifier(transientProject.getClassifier());
+        project.setCollectionLogic(transientProject.getCollectionLogic());
```

Comment:
That's correct, but also it would be an intended change. Idk what could be done about this except add such change to the logfile, but that would only be helpful for deep analysis of such issue.

Reply:
Fair.

Naive idea: We could add a *Collection logic change* flag to `ProjectMetrics`. We could use that in the metrics graph, in some similar way to what you can do in Grafana with annotations:

<img width="477" alt="image" src="https://github.com/DependencyTrack/dependency-track/assets/5693141/e422f3cb-51dc-482a-bf9d-127b7a6edf29">

Reply:
I like that idea, but at least on a first look the frontend chart lib doesn't support this. not sure how difficult this would be to add on top of it. Any experience with that? I never used that component.
From data perspective in backend it should not be too complicated.

Reply:
I don't have *too* strong opinions on *how* we highlight it, but I do think that highlighting it one way or the other would be good.

Tooltip on it's own might be too hidden, but styling the datapoints sounds good.

Reply:
@nscuro would you be fine with creating a new issue to add this visibility later as further enhancement, and for now rely on logile for visibility in case someone wonders? we could also add a hint in the project edit UI when someone changes the logic, that informs him of this for now?

Reply:
Logging is probably fine for an MVP implementation, yes. But ultimately we will really need some kind of indicator in the UI.

Reply:
@nscuro  I found a way to show it in the metrics chart. Only issue is that the line gets invisible when hovered. No Idea why. But I think this is fine as it is, looks like its intended :D

Reply:
Ok found the reason for the visibility issue and fixed it. Also extended it to the other charts now. Looks like this:

![image](https://github.com/user-attachments/assets/d6c8db14-5530-497e-9759-7b937d700641)